File: connectors/hybris-dxa-module/src/main/java/com/sdl/ecommerce/hybris/api/model/Category.java
package com.sdl.ecommerce.hybris.api.model;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import javax.annotation.Generated;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Generated("org.jsonschema2pojo")
@JsonPropertyOrder({"code", "url", "image"})
public class Category {

    @JsonProperty("id")
    private String id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("url")
    private String url;

    @JsonProperty("image")
    private Image image;

    @JsonProperty("subcategories")
    private List<Category> subcategories;

    @JsonProperty("products")
    private List<Product> products;

    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getUrl() {
        return url;
    }

    public Image getImage() {
        return image;
    }

    public List<Category> getSubcategories() {
        return subcategories;
    }

    public List<Product> getProducts() {
        return products;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: ecl/hybris-ecl-provider/hybris-ecl-provider/Hybris API/Category.cs
namespace SDL.ECommerce.Hybris.API.Model
{
    using System;
    using System.Collections.Generic;
    using System.Diagnostics.CodeAnalysis;

    /// <summary>
    /// Category class
    /// </summary>
    [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1300:ElementMustBeginWithUpperCaseLetter", Justification = "auto generated class needs to map to json file")]
    public class Category
    {
        public string code { get; set; }
        public string id { get; set; }
        public DateTime lastModified { get; set; }
        public string name { get; set; }
        public string url { get; set; }
        public List<Category> subcategories { get; set; }
        public List<Product> products { get; set; }
        public Category Parent { get; set; }
    }
}
